src/tests/ftest/recovery/pass_1.py

```diff
@@ -0,0 +1,416 @@
+    """Test Pass 1: Pool List Consolidation
+
+    :avocado: recursive
+    """
+
+    def test_dangling_pool(self):
+        """Test dangling pool.
+
+        1. Create a pool.
+        2. Remove the pool from the pool service (PS) on engine by calling:
+        dmg faults pool-svc <pool> CIC_POOL_NONEXIST_ON_ENGINE
+        3. Show dangling pool entry by calling:
+        dmg pool list --no-query
+        4. Enable and start the checker.
+        5. Query the checker and verify the message regarding dangling pool.
+        6. Disable the checker.
+        7. Verify that the dangling pool was removed. Call dmg pool list and it should
+        return an empty list.
+        8. Un-register the pool so that we don't try to destroy during tearDown.
+
+        Jira ID: DAOS-11711
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=recovery,pass_1
+        :avocado: tags=test_dangling_pool
+        """
+        # 1. Create a pool.
+        self.pool = self.get_pool(connect=False)
+
+        # 2. Remove the pool from the pool service (PS) on engine.
+        dmg_command = self.get_dmg_command()
+        dmg_command.faults_pool_svc(
+            pool=self.pool.identifier, checker_report_class="CIC_POOL_NONEXIST_ON_ENGINE")
+
+        # 3. Show dangling pool entry.
+        pools = dmg_command.get_pool_list_labels(no_query=True)
+        if self.pool.identifier not in pools:
+            self.fail("Dangling pool not found after removing pool from PS!")
+
+        # 4. Enable and start the checker.
+        dmg_command.check_enable()
+        # Calling dmg check start will automatically fix the issue with the default
+        # option, which is to remove the dangling pool in this failure type.
+        dmg_command.check_start()
+
+        # 5. Query the checker.
+        query_msg = ""
+        for _ in range(8):
+            check_query_out = dmg_command.check_query()
+            if check_query_out["response"]["status"] == "COMPLETED":
+                query_msg = check_query_out["response"]["reports"][0]["msg"]
+                break
+            time.sleep(5)
+
+        # Verify that the checker detected dangling pool.
+        errors = []
+        if "dangling pool" not in query_msg:
+            errors.append(
+                "Checker didn't detect dangling pool! msg = {}".format(query_msg))
+
+        # 6. Call dmg check disable.
+        dmg_command.check_disable()
+
+        # 7. Verify that the dangling pool was removed.
+        pools = dmg_command.get_pool_list_labels()
+        if pools:
+            errors.append(f"Dangling pool was not removed! {pools}")
+
+        # 8. Un-register the pool so that we don't try to destroy during tearDown.
+        # This is hack because when we create a pool using get_pool(), it'll call
+        # register_cleanup() and there's no way to skip it or "unregister" other than this
+        # approach.
+        self._cleanup_methods = []
+
```

Comment:
Have you looked at `pool.skip_cleanup()`? Modifying `self._cleanup_methods` directly is not a good idea

Reply:
I changed it to use `skip_cleanup()`. Thanks for letting me know!